src/IECoreScene/SceneCache.cpp

```diff
@@ -728,8 +730,76 @@ class SceneCache::ReaderImplementation : public SceneCache::Implementation
 			return reader;
 		}
 
+		ConstPathMatcherDataPtr readSet( const Name &name ) const
+		{
+			IECore::PathMatcherDataPtr pathMatcherData = readLocalSet( name );
+
+			NameList children;
+			childNames( children );
+
+			for( const auto &c : children )
+			{
+				ReaderImplementation *nc = const_cast<ReaderImplementation *>(this);
```

Comment:
This is a variation of the const problem I had with the SceneInterface getRoot.

Reply:
Ah, so this is because `ReaderImplementation::child()` is non-const? What happens if we make it `const`?